src/tactic/core.lean

```diff
@@ -360,6 +360,26 @@ meta def lambdas : list expr → expr → tactic expr
   pure $ expr.lam pp info t (expr.abstract_local f' uniq)
 | _ f := pure f
 
+/--  Given an expression `f` (likely a binary operation) and a further expression `x`, calling
+`list_binary_operands f x` breaks `x` apart into successions of applications of `f` until this can
+no longer be done and returns a list of the leaves of the process.
```

Comment:
```suggestion
no longer be done and returns a list of the leaves of the process.

This matches `f` up to semiredicible unification. In particular it will match applications of the same polymorphic function with different type class arguments. E.g., if `i1` and `i2` are both instances of `has_add T` and `e := has_add.add T i1 x (has_add.add T i2 y z)`, then ``list_binary_operands `((+) : T → T → T) e`` will return `[x, y, z]`.
```